README.md

```diff
@@ -76,6 +76,28 @@ It doesn't hurt to not do this as squishy assumes 30 second visibility timeouts
 by default. It would just result in more SQS API calls than are strictly
 necessary.
 
+### Very Long Running Jobs
+
+If you sometimes/often have jobs that run over the maximum visibility timeout
+for SQS (12 hours), you will get new copies of the message spawning without
+bounds every 12 hours until you manually intervene. Since this is undesireable,
+there is an option you can provide to `consume-messages` called `:delete-callback`.
+When set to true, your processing function will receive two arguments, the
+message to process and a callback function to be called when enough information has
+been stored to keep track of the job. Calling this function will then delete the
+message from SQS and no copies will spawn (presuming you have called this prior
+to the 12 hour maximum visibility timeout).
```

Comment:
This seems to be what the `timeout-increaser` future does - implement the ["heartbeat" for the consumer process](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html#configuring-visibility-timeout) pattern.

Since that `ChangeMessageVisibility` call clearly isn't capped at [43200](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ChangeMessageVisibility.html#API_ChangeMessageVisibility_RequestParameters), could that have been failing after a certain period of time which then caused new copies of the message to spawn?

Reply:
@gws Yeah, basically if you attempt to go over the maximum visibility for any message, 43200, I think it silently agrees (or we're not paying attention to the response indicating failure) but it doesn't actually extend anything. So after 12 hours we see new spawned copies and effectively DOS ourselves after a day.

The `timeout-increaser` probably still has a place in the world where you want somewhat fast moving messages and just sometimes need to bump it up once or twice. What I'm proposing is more or less for our feed processing needs where 12+ hours in uncommon but expected and we need a way to kill the message from anyone else picking it up once we've progressed far enough to record information about it.